contracts/LimitOrderSwap.sol

```diff
@@ -92,6 +92,10 @@ contract LimitOrderSwap is ILimitOrderSwap, Ownable, TokenCollector, EIP712 {
                 if (makingAmount > orderAvailableAmount) revert NotEnoughForFill();
                 takerTokenAmounts[i] = ((makingAmount * order.takerTokenAmount) / order.makerTokenAmount);
 
+                if (order.takerToken == address(weth)) {
+                    wethToPay += takerTokenAmounts[i];
+                }
```

Comment:
I think these block could be placed after zero amount check (line 99)

Reply:
Fixed in 0708ce1